File: csharp/p9.cs
using System;

class P9{
  static void Main(string[] args){
    uint sum = 1000;
    if(args.Length >= 1){
      sum = uint.Parse(args[0]);
    }
    uint prod = 0;
    for(uint a = 3; a < (sum-3)/3 && prod==0; ++a){
      for(uint b = (a+1); b<(sum-1-a)/2 && prod == 0; ++b){
        uint c = sum-a-b;
        if(c*c == a*a+b*b){
          prod = a*b*c;
        }
      }
    }
    Console.Write("{0}\n", prod);
  }
}

File: rust/p9.rs
use std::env;

fn main() {
    let mut sum: i64 = 1000;
    let args: Vec<_> = env::args().collect();
    if args.len() > 1 {
        sum = args[1].parse().unwrap();
    }
    let mut prod: i64 = 0;
    for a in 3..(sum-3)/3 {
        for b in (a+1)..(sum-1-a)/2 {
            let c: i64 = sum-a-b;
            if c * c == a * a + b * b {
                prod = a * b * c;
            }
        }
    }
    println!("{}", prod);
}
